src/main/resources/data/computercraft/lua/rom/apis/rednet.lua

```diff
@@ -414,14 +414,15 @@ function run()
                 end
             end
 
-        elseif sEvent == "timer" then
+        elseif sEvent == "timer" and p1 == nClearTimer then
             -- Got a timer event, use it to clear the event queue
-            local nTimer = p1
-            local nMessage = tReceivedMessageTimeouts[nTimer]
-            if nMessage then
-                tReceivedMessageTimeouts[nTimer] = nil
-                tReceivedMessages[nMessage] = nil
+            nClearTimer = nil
+            local nNow, bFound = os.epoch("utc"), nil
+            for k, v in pairs(tReceivedMessages) do
+                if nNow - v >= 9500 then tReceivedMessages[k] = nil
```

Comment:
Following on from my `os.clock` message above, then you'd just have:

```suggestion
                if deadline >= now then tReceivedMessages[k] = nil
```